python/cugraph-pyg/cugraph_pyg/nn/conv/transformer_conv.py

```diff
@@ -0,0 +1,210 @@
+        heads: int = 1,
+        concat: bool = True,
+        beta: bool = False,
+        edge_dim: Optional[int] = None,
+        bias: bool = True,
+        root_weight: bool = True,
+    ):
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.heads = heads
+        self.beta = beta and root_weight
+        self.root_weight = root_weight
+        self.concat = concat
+        self.edge_dim = edge_dim
+
+        if isinstance(in_channels, int):
+            in_channels = (in_channels, in_channels)
+
+        self.lin_key = nn.Linear(in_channels[0], heads * out_channels)
+        self.lin_query = nn.Linear(in_channels[1], heads * out_channels)
+        self.lin_value = nn.Linear(in_channels[0], heads * out_channels)
+
+        if edge_dim is not None:
+            self.lin_edge = nn.Linear(edge_dim, heads * out_channels, bias=False)
+        else:
+            self.lin_edge = self.register_parameter("lin_edge", None)
+
+        if concat:
+            self.lin_skip = nn.Linear(in_channels[1], heads * out_channels, bias=bias)
+            if self.beta:
+                self.lin_beta = nn.Linear(3 * heads * out_channels, 1, bias=bias)
+            else:
+                self.lin_beta = self.register_parameter("lin_beta", None)
+        else:
+            self.lin_skip = nn.Linear(in_channels[1], out_channels, bias=bias)
+            if self.beta:
+                self.lin_beta = nn.Linear(3 * out_channels, 1, bias=False)
+            else:
+                self.lin_beta = self.register_parameter("lin_beta", None)
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        self.lin_key.reset_parameters()
+        self.lin_query.reset_parameters()
+        self.lin_value.reset_parameters()
+        if self.lin_edge is not None:
+            self.lin_edge.reset_parameters()
+        if self.lin_skip is not None:
+            self.lin_skip.reset_parameters()
+        if self.lin_beta is not None:
+            self.lin_beta.reset_parameters()
+
+    def forward(
+        self,
+        x: Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]],
+        csc: Tuple[torch.Tensor, torch.Tensor, int],
+        edge_attr: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        r"""Runs the forward pass of the module.
+
+        Args:
+            x (torch.Tensor or tuple): The node features. Can be a tuple of
+                tensors denoting source and destination node features.
+            csc ((torch.Tensor, torch.Tensor, int)): A tuple containing the CSC
+                representation of a graph, given as a tuple of
+                :obj:`(row, colptr, num_src_nodes)`. Use the
+                :meth:`to_csc` method to convert an :obj:`edge_index`
+                representation to the desired format.
+            edge_attr: (torch.Tensor, optional) The edge features.
+        """
+        bipartite = not isinstance(x, torch.Tensor)
```

Comment:
```suggestion
        bipartite = isinstance(x, Tuple)
```

Reply:
This was intended for also accepting a pair in list.